File: demo/app/src/main/java/com/demo/ui/auth/viewmodel/factory/LoginViewModelFactory.java
package com.demo.ui.auth.viewmodel.factory;

import androidx.annotation.NonNull;
import androidx.lifecycle.ViewModel;
import androidx.lifecycle.ViewModelProvider;
import com.demo.data.UserRepository;
import com.demo.ui.auth.viewmodel.LoginViewModel;

public class LoginViewModelFactory implements ViewModelProvider.Factory {

    private UserRepository mUserRepository;

    public LoginViewModelFactory(UserRepository userRepository) {
        this.mUserRepository = userRepository;
    }

    @NonNull
    @Override
    public <T extends ViewModel> T create(@NonNull Class<T> modelClass) {

        LoginViewModel loginViewModel = new LoginViewModel(mUserRepository);

        return (T) loginViewModel;
    }
}

File: demo/appko/src/main/java/com/demo/appko/ui/auth/viewmodel/LoginViewModelFactory.kt
package com.demo.appko.ui.auth.viewmodel

import androidx.lifecycle.ViewModel
import androidx.lifecycle.ViewModelProvider
import com.demo.appko.data.UserRepository

public class LoginViewModelFactory(private val userRepository: UserRepository) : ViewModelProvider.Factory{

    override fun <T : ViewModel?> create(modelClass: Class<T>): T {
        if (modelClass != LoginViewModel::class.java) {
            throw IllegalArgumentException("Unknown ViewModel class")
        }
        val viewModel = LoginViewModel(userRepository)
        return viewModel as T
    }

}
